driver/server/grpc.go

```diff
@@ -58,32 +58,56 @@ func newDriverManifest(manifest *manifest.Manifest) protocol1.DriverManifest {
 	}
 }
 
+// SupportedLanguages implements protocol1.Service.
 func (s service) SupportedLanguages(_ *protocol1.SupportedLanguagesRequest) *protocol1.SupportedLanguagesResponse {
-	m, _ := s.d.Manifest()
-	return &protocol1.SupportedLanguagesResponse{Languages: []protocol1.DriverManifest{
-		newDriverManifest(&m),
-	}}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	list, err := s.d.Languages(ctx)
+	if err != nil {
+		return &protocol1.SupportedLanguagesResponse{Response: errResp(err)}
+	}
+	resp := &protocol1.SupportedLanguagesResponse{
+		Response: protocol1.Response{Status: protocol1.Ok},
+	}
+	for _, m := range list {
+		resp.Languages = append(resp.Languages, newDriverManifest(&m))
+	}
+	return resp
 }
 
 func (s service) parse(mode driver.Mode, req *protocol1.ParseRequest) (nodes.Node, protocol1.Response) {
 	start := time.Now()
-	m, err := s.d.Manifest()
+	ctx := context.Background()
+	if req.Timeout > 0 {
+		var cancel func()
+		ctx, cancel = context.WithTimeout(ctx, req.Timeout)
+		defer cancel()
+	}
+	list, err := s.d.Languages(ctx)
 	if err != nil {
 		r := errResp(err)
 		r.Elapsed = time.Since(start)
 		return nil, r
 	}
-	if req.Language != m.Language {
+	ok := false
```

Comment:
Suggestion:

```go
func containsLang(ms []*manifest.Manifest, lang string) bool {
	for _, m := range ms {
		if m.Language == lang {
			return true
		}
		for _, alias := range m.Aliases {
			if alias == lang {
				return true
			}
		}
	}
	return false
}

…
if !containsLang(list, req.Language) {
   …
```

I know we only use it once, but having to use state variables to track control flow always makes me nervous 😄.